### DISCUSSION TITLE
Problem configuring the jdtls server

### DISCUSSION THREAD

**Discussion by BitInByte** [mfussenegger/nvim-jdtls#175] [Q&A] [answered]
Hello everyone, hope's everyone well and safe...

I've been trying to implement nvim-jdtls and I made so much stuff that now I have 2 jdtls launching and I get a error when trying to implement a code_action.

My config right now looks like this:

<img width="1245" alt="Screenshot 2021-11-09 at 13 48 16" src="https://user-images.githubusercontent.com/64453275/140935564-1dfbb4d7-5284-4501-879a-4a657195fa14.png">

and my .zshenv:
`
export JDTLS_HOME=/Library/Java/jdt-language-server-1.5.0-202110191539 
export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk-11.0.12.jdk/Contents/Home/ 
export WORKSPACE=~/.cache/jdtls-workspace # Defaults to $HOME/workspace  
`

And my :LspInfo looks like this:
<img width="1158" alt="Screenshot 2021-11-09 at 13 45 44" src="https://user-images.githubusercontent.com/64453275/140935224-e94daf1d-3aaa-408b-8fd7-28b8580bbc93.png">

And if I try to run for example generate toString()... I get the following error:
` -32601: No delegateCommandHandler for java.action.generateToStringPrompt`

Hope's someone can help me with this because I'm really struggling and thanks in advance :/.

**Comment by mfussenegger** [comment]
Do you still have a `require'lspconfig'.jdtls.setup{}` somewhere? Can you try removing that. It leads to the second server getting started.

The `nvim-jdtls` specific config you shared looks okay on a first glance.

**Reply by BitInByte** [reply]
That's the strange thing, I don't have `require'lspconfig'.jdtls.setup{}` nowhere. I re-build my config entirely yesterday because I switched from Vimscript to lua config and I don't have it somewhere else, just on this file that you saw. I'm so struggling :/

**Reply by foxdefox-wq** [reply]
it's lspconfig that doing it, you just gotta find a way to disable it autosetting up jdtls 
I don't really know how to yet

**Comment by BitInByte** [comment, marked as answer]
Solve one of the issues but there's another one, the error: 
`-32601: No delegateCommandHandler for java.action.generateToStringPrompt`

**Reply by BitInByte** [reply]
Ok, for record, if anyone got's this error, the reason is that you need to use the jdtls code_action function and not the Neovim LSP code_action function. Now I think that everything it's up and running @mfussenegger, thank you for the amazing work and support :)

**Reply by mfussenegger** [reply]
Glad you managed to get it working. The thing about jdtls code_action vs. neovim code_action btw. won't be necessary anymore with the upcoming neovim 0.6

**Reply by BitInByte** [reply]
Really? That's awesome... Basically what happened to me was I didn't remap the keys to the jdtls functions and I had a for which was initialising a couple of LSPs and one of them was jdtls. Now seems to work.

Just 2 more questions @mfussenegger, the autostart it is supposed to be false? And, it is normal that the plugin creates a workplace directory in the project root directory?

Thank you so much :)

**Reply by mfussenegger** [reply]
>  the autostart it is supposed to be false? 

yes that's normal. `:LspInfo` is a command provided by lspconfig and jdtls has currently no good way to hook into that. But given that both jdtls and lspconfig use the underlying neovim LSP client it shows up anyways. 

There are some plans to move more of the "manager" parts of lspconfig into neovim core and then that might change, but too soon to tell if that will work differently then or not.

> is normal that the plugin creates a workplace directory in the project root directory?

the eclipse.jdt.ls language server creates a `.project` file and a `.settings` folder in the project root dir. That's normal.

**Reply by BitInByte** [reply]
In my case it is creating a .settings folder, a .project file and a workspace folder in the root of my project directory @mfussenegger 
<img width="273" alt="Screenshot 2021-11-09 at 19 49 02" src="https://user-images.githubusercontent.com/64453275/140994468-4877008d-18dc-4d3d-af26-005af19e37d8.png">

**Reply by mfussenegger** [reply]
No idea where that `workspace` folder is coming from. I don't have that. 

There is btw. work ongoing to no longer create metadata files in the project root: https://github.com/eclipse/eclipse.jdt.ls/pull/1900
